.github/workflows/sql-release.yml

```diff
@@ -56,6 +72,12 @@ jobs:
         run: |
           rm sql/target/*.txt
           zip -j librillsql-${{ matrix.os }}-${{ matrix.arch }}.zip sql/target/librillsql.* sql/target/graal_isolate.*
+      - name: Push version changes
+        uses: EndBug/add-and-commit@v7
+        with:
+          default_author: github_actions
+          add: sql/pom.xml
+          message: librillsql release v${{ steps.maven-version.outputs.version }}
```

Comment:
This is going to cause a push directly to `main`, which feels a little dangerous – we would prefer always to have manual merges to `main`. I can see a few options:

- Either run this once per PR and commit the version update to the PR. This has the risk that two simultaneous PRs will update to the same version, which requires manual intervention.
- Or, have the developer manually bump the version (put the command in `sql/README.md` for convenience) and then have CI check that a pull request does not have the same version number as the one on `main` (so we don't forget)
- Or, don't automatically release on merges to main – come up with another release approach (like a manually triggered workflow that checks that the version number has been updated)
- Or, just use SHAs as version numbers

Any other ideas?

Reply:
I don't think it's more dangerous than committing manually. But as a second option I can propose manual version change by PR committer.